treeherder/etl/buildapi.py

```diff
@@ -307,6 +334,13 @@ def transform(self, data, source, filter_to_revision=None, filter_to_project=Non
                 # using project and revision form the revision lookups
                 # to filter those jobs with unmatched revision
                 for job in jobs:
+                    job_ids_seen_now.add(job['id'])
```

Comment:
Actually this won't work, since we're already inside the revision loop, so we'll have to batch the job ids up from each non-skipped revision anyway. Given this, plus post bug 1162947 it may end up having to be removed, I'll leave as-is for now, but as part of bug 1162947 I'll take a close look so see how we might refactor these loops in general :-)